ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -147,12 +168,55 @@ static StatsSnapshot convertQuotaToStatsSnapshot(Map<String, Map<String, Long>>
 
   /**
    * {@inheritDoc}
-   * Implementation in {@link BlobStoreStats} which returns the quota stats of a {@link BlobStore}. Size of delete
-   * records are not accounted as valid data size here.
+   * Implementation in {@link BlobStoreStats} which returns the all types of snapshots for a {@link BlobStore}. Size of
+   * delete records are not accounted as valid data size here. The formats of all snapshots are presented as follows.
+   * <pre>
+   *        AccountSnapshot             |             ContainerSnapshot
+   * --------------------------------------------------------------------------
+   * {                                  |      {
+   *   value: 1000,                     |        value: 1000,
+   *   subMap: {                        |        subMap: {
+   *     Account[1]:{                   |           Account[1]_Container[1]: {
+   *       value: 200,                  |               value: 200,
+   *       subMap: {                    |               subMap: null
+   *         Container[1]:{             |           },
+   *           value: 200,              |           Account[2]_Container[3]: {
+   *           subMap: null             |               value: 800,
+   *         }                          |               subMap: null
+   *       }                            |           }
+   *     },                             |        }
+   *     Account[2]:{                   |      }
+   *       value: 800,                  |
+   *       subMap:{                     |
+   *         Container[3]:{             |
+   *           value: 800,              |
+   *           subMap: null             |
+   *         }                          |
+   *       }                            |
+   *     }                              |
+   *   }                                |
+   * }                                  |
+   * </pre>
    */
   @Override
-  public StatsSnapshot getStatsSnapshot(long referenceTimeInMs) throws StoreException {
-    return convertQuotaToStatsSnapshot(getValidDataSizeByContainer(referenceTimeInMs));
+  public Map<StatsReportType, StatsSnapshot> getStatsSnapshots(EnumSet<StatsReportType> statsReportTypes,
+      long referenceTimeInMs) throws StoreException {
+    Map<StatsReportType, StatsSnapshot> statsSnapshotsByType = new HashMap<>();
+    Map<String, Map<String, Long>> quotaMap = getValidDataSizeByContainer(referenceTimeInMs);
+    for (StatsReportType reportType : statsReportTypes) {
```

Comment:
I have a couple of terminology questions:
1. Is there a difference between a "stats report" and a "stats snapshot"?
2. Why is the storage usage referred to as a quota? Typically a quota refers to the actual limit set on something, not the usage against the quota.

Reply:
Good question. Let me try to answer it. (we can discuss offline if there is a need to rename them)
1. `stats snapshot` is an intermediate result generated by each container or account in each partition. `stats report` is the final result generated after aggregating stats snapshots from all partitions on a single node. 
2. `quota` comes from legacy code. Actually when I was writing this code, I felt like it was a bit of inappropriate because it had conflict with the Nuage term. Maybe we should rename it to usage instead. What do you think?

Reply:
Maybe I should make it clear in the code. `AccountSnapshot` is only used for generating final `ACCOUNT_REPORT` and `ContainerSnapshot` is only used for `PARTITION_CLASS_REPORT`.

Reply:
Thanks! I think it is more understandable now.